tests/suites/test_suite_pk.function

```diff
@@ -19,12 +19,21 @@
 #include "psa/crypto.h"
 #include "mbedtls/psa_util.h"
 
+#include <test/psa_exercise_key.h>
+
 /* Used for properly sizing the key buffer in pk_genkey_ec() */
 #include "psa_util_internal.h"
 
 #define RSA_KEY_SIZE   MBEDTLS_RSA_GEN_KEY_MIN_BITS
 #define RSA_KEY_LEN   (MBEDTLS_RSA_GEN_KEY_MIN_BITS/8)
 
+#if defined(MBEDTLS_RSA_C) ||                                           \
+    defined(MBEDTLS_PK_RSA_ALT_SUPPORT) ||                              \
+    defined(MBEDTLS_ECDSA_C) ||                                         \
+    defined(MBEDTLS_USE_PSA_CRYPTO)
```

Comment:
The list here matches the conditions for `MBEDTLS_PK_SIGNATURE_MAX_SIZE` to be nonzero, based on how it's set in `include/mbedtls/pk.h`. That was the motivation for 74860ddec2486ea63316d77cd4a3f3146c127bda.

We can't remove `MBEDTLS_RSA_C` or `MBEDTLS_PK_RSA_ALT_SUPPORT` since these should be enabled even when PSA is disabled. We can't remove `MBEDTLS_USE_PSA_CRYPTO` because this covers `MBEDTLS_PK_OPAQUE` which can reach PSA-only signature algorithms.

Arguably we should replace `defined(MBEDTLS_ECDSA_C)` by `MBEDTLS_PK_CAN_ECDSA_SIGN`? I'm not sure it makes a difference in terms of maintainability. Intuitively, to me:

* Either `MBEDTLS_USE_PSA_CRYPTO` is enabled, and this is always on: PK can at least try to sign through PSA (although it's possible that PSA doesn't actually support any signature algorithm, but we currently don't need to handle this refinement).
* Or `MBEDTLS_USE_PSA_CRYPTO` is disabled, and then we know that PK is using the built-in features for anything other than `MBEDTLS_PK_OPAQUE`. All the built-in features that can sign are listed here.

By the way this is one of many places that need to change when `MBEDTLS_PK_OPAQUE` is generalized to work without `MBEDTLS_USE_PSA_CRYPTO`. I don't know of an easy way to find them all.